tests/unit/test_container.cc

```diff
@@ -97,6 +97,22 @@ static void testContainer(bool is_ordered, std::initializer_list<T> lst) {
   }
 }
 
+template <>
+void testContainer<std::vector<int>, int>(
+  bool /*is_ordered*/, std::initializer_list<int> lst
+) {
+  std::vector<int> c1{lst};
+  c1.reserve(1000);
+  auto ret = checkpoint::serialize(c1);
+
+  auto t1 = checkpoint::deserialize<std::vector<int>>(ret->getBuffer());
+
+  EXPECT_EQ(c1.size(), t1->size());
+  EXPECT_EQ(c1.capacity(), t1->capacity());
```

Comment:
The reserved capacity should be a named constant, and the comparison should be that the deserialized vector has capacity greater than or equal to both the constant and the original vector. The call to `reserve` can over allocate at both sites

Reply:
fixed, explanation added in the comment